rest/replication_api_test.go

```diff
@@ -1081,17 +1058,30 @@ func TestValidateReplication(t *testing.T) {
 				QueryParams: []string{"E", "A", "D", "G", "B", "e"},
 				Direction:   db.ActiveReplicatorTypePull,
 			},
-			errExpected: &base.HTTPError{
-				Status:  http.StatusBadRequest,
-				Message: "Bad channels array in query_params for sync_gateway/bychannel filter",
+			expectedErrorMsg: db.ConfigErrorBadChannelsArray,
+		},
+		{
+			name: "replication config replicationID too long",
+			replicationConfig: db.ReplicationConfig{
+				ID: "0123456789012345678901234567890123456789012345678901234567890123456789" +
+					"0123456789012345678901234567890123456789012345678901234567890123456789" +
+					"0123456789012345678901234567890123456789012345678901234567890123456789",
+				Remote: "http://bob:pass@remote:4984/db",
 			},
+			expectedErrorMsg: db.ConfigErrorIDTooLong,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			err := tc.replicationConfig.ValidateReplication(tc.fromConfig)
-			assert.Equal(t, tc.errExpected, err)
+			if tc.expectedErrorMsg != "" {
+				expectedError := &base.HTTPError{
+					Status:  http.StatusBadRequest,
+					Message: tc.expectedErrorMsg,
+				}
+				assert.Equal(t, expectedError, err)
+			}
```

Comment:
To cover the happy-path test cases

```suggestion
			} else {
				assert.NoError(t, err)
			}
```